tycho-indexer/src/extractor/protocol_extractor.rs

```diff
@@ -576,6 +586,21 @@ where
         let mut msg =
             if let Some(post_process_f) = self.post_processor { post_process_f(msg) } else { msg };
 
+        if let Some(last_processed_block) = self.get_last_processed_block().await {
+            if msg.block.ts.timestamp() == last_processed_block.ts.timestamp() {
+                // Blockchains with fast block times (e.g., Arbitrum) may produce blocks with
+                // identical timestamps. To ensure accurate ordering, we adjust each
+                // block's timestamp by adding a microsecond offset based on the
+                // number of blocks with the same timestamp encountered so far.
+                msg.block.ts += chrono::Duration::microseconds(
+                    last_processed_block
+                        .ts
+                        .timestamp_subsec_micros() as i64 +
+                        1,
+                );
```

Comment:
This could be simplified a lot no?
```suggestion
                msg.block.ts = last_processed_block.ts + Duration::microseconds(1);
```

Reply:
I also find this more natural

Reply:
Another question here though... why microseconds instead of milliseconds? What granularity do we store in the db? I thought it was only to milliseconds?

Reply:
ohh - nice one, tnx!